blog/2021-12-29-auto-setup.md

```diff
@@ -0,0 +1,121 @@
+2. Run the `setup-schema` command with the `temporal-[db]-tool` 
+3. Run the `update-schema` command with the `temporal-[db]-tool` with a specific schema for that combination of database + schema version  
+4. Create a new Standard Visibility database with `VISIBILITY_DBNAME`
+5. Run the `setup-schema` command with the `temporal-[db]-tool`
+6. Run the `update-schema` command with the `temporal-[db]-tool` with a specific Visibility schema for that combination of database + schema version  
+
+If enabled, Temporal’s Advanced Visibility Schema setup [works in a similar way](https://github.com/temporalio/temporal/tree/master/schema/elasticsearch/visibility) as described above.
+
+## Temporal Server Setup
+
+Once the schema is setup, the `setup_server` function is started in a background. This function waits for Temporal server to get started which happens in a meantime in the main process (in `start-temporal.sh`). When Temporal server is started, this function does one more important thing that users may take for granted - it registers the `default` namespace using the standard `tctl --ns default namespace register` command. 
+
+- This is the namespace that most beginners are used to seeing when they start up Temporal inside docker-compose - because the `auto-setup.sh` script does it for them! Without this script running, Temporal Server doesn’t have any namespaces registered. This can come as a surprise to some users who assume the default namespace is a necessary part of Temporal.
+- All SDKs Workers and Clients, as well as Temporal Web, connect to the `default` namespace if not specified - most production deployments of Temporal will want to connect to specific namespaces. (try running [temporalite](https://github.com/DataDog/temporalite) without any namespaces to see how SDK code without namespaces specified stops working!)
+- The script allows you to customize the name and retention period with the `DEFAULT_NAMESPACE` and `DEFAULT_NAMESPACE_RETENTION` env vars if you wish, but a production setup may want to register a number of namespaces differently.
+
+The final part of the `auto-setup.sh` script also does registers some basic custom search attributes:
+
+```tsx
+tctl --auto_confirm admin cluster add-search-attributes \
+          --name CustomKeywordField --type Keyword \
+          --name CustomTextField --type Text \
+          --name CustomIntField --type Int \
+          --name CustomDatetimeField --type Datetime \
+          --name CustomDoubleField --type Double \
+          --name CustomBoolField --type Bool
+```
+
+But these are for demo purposes (eg [for code samples](https://github.com/temporalio/samples-go/blob/77728cf7c38570898b2c90bf6eb0720c7f5fb30d/searchattributes/searchattributes_workflow.go#L56-L63)) rather than illustrative of real usage - hence the extremely generic naming of these attributes. In practice, you are likely to drop this in favor of manual or custom setup of search attributes.
```

Comment:
Don't actually think thats true. These are nice because there is a limit on search attributes and its a pain to add new ones for some people.

Reply:
```suggestion
These are intentionally generically named, for demo purposes (eg [for code samples](https://github.com/temporalio/samples-go/blob/77728cf7c38570898b2c90bf6eb0720c7f5fb30d/searchattributes/searchattributes_workflow.go#L56-L63)) and for ease of use (since they are already set up, you don't have to add them when you find you need them later). As you advance in your usage, you may wish to drop this step in favor of better named and specified attributes - just keep in mind that you are [limited to a maximum of 100](https://docs.temporal.io/docs/server/production-deployment/#server-limits).
```